x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -121,24 +113,6 @@ public AuthenticationService getAuthenticationService() {
         return authenticationService;
     }
 
-    private CrossClusterAccessSubjectInfo maybeRewriteForCrossClusterAccessUser(
-        AuthenticationService.AuditableRequest request,
-        CrossClusterAccessSubjectInfo crossClusterAccessSubjectInfo
-    ) {
-        final Authentication authentication = crossClusterAccessSubjectInfo.getAuthentication();
-        final Subject effectiveSubject = authentication.getEffectiveSubject();
-        final User user = effectiveSubject.getUser();
-        if (CrossClusterAccessUser.is(user)) {
-            logger.debug("Request [{}] performed by internal user [{}]. Will use pre-defined role descriptors", request, user.principal());
-            return CrossClusterAccessUser.subjectInfoWithRoleDescriptors(
```

Comment:
This had the side-effect that e.g., extra metadata was stripped. I think it makes sense to leave it out here, and instead sanitize all authentication instances the FC receives (esp. w.r.t. metadata). That way, we protect against bogus metadata more generally. WDYT?

Reply:
For sure, we can. It just keeps an extra layer of translation that makes the code slightly more complicated. I don't feel strongly though. Will add it back.

Reply:
Appologies, I think I misunderstood. You meant to drop it. I thought you wanted to keep it. My suggestion was to use `subjectInfoWithEmptyRoleDescriptors` if we are going to keep it. If you are going to drop it, my suggestion is invalid and I am OK with dropping it too. Sorry about the confusion.

Reply:
Yup sorry my intention was to drop it, and potentially add a more general sanitization step that removes "un-allow-listed" metadata from received authentication instances (in a separate PR, if at all).